### DISCUSSION TITLE
Advanced search page

### DISCUSSION THREAD

**Discussion by triphora** [modrinth/.github#15] [Feature requests]
The current frontend has no way of supporting `OR` and `NOT` filtering.
It's been brought up a couple times in private channels that an "advanced search" page is going to be a necessity for allowing the powerful [search API](https://docs.modrinth.com/docs/tutorials/api_search/) to be used to its fullest extent in the frontend.
![super old mockup](https://cdn.discordapp.com/attachments/848329986577399818/856770306240413716/Generic_search2x.png) ([source](https://canary.discord.com/channels/734077874708938864/848329986577399818/856770308219600896) - @venashial in `#contributors`)
Most recent discussion [here](https://canary.discord.com/channels/734077874708938864/848329986577399818/948054916205789184) (also private channel, sorry passersby)

**Comment by Minenash** [comment]
One nice thing to have in a text-only search, would be the ability to do `1.18.x` and `1.18.s`(or `1.18-snapshots`), as typing `1.18,1.18.1,1.18.2` is more tedious then selecting it (I mean I guess this applies to everything, but it's a lot more in this case)

**Reply by Secret-chest** [reply]
Maybe `1.18.*`

**Comment by intergrav** [comment]
Concept I made in Figma for the modern site

![image](https://user-images.githubusercontent.com/42325132/201506969-2dc672e7-623a-4dc3-bde2-b2639f5c6594.png)

**Comment by awebeer256** [comment]
Two years later, has there been any progress on this?

Right now I'm _really_ feeling the inability to OR-combine loaders and categories in mod search, e.g. "Fabric or Quilt."

**Reply by Prospector** [reply]
Loaders do use OR, so Fabric or Quilt works

**Reply by awebeer256** [reply]
Yeah, that's on me :facepalm:. Minecraft versions also use OR, as it turns out. So let me revise: I'm really feeling the inability to OR-combine categories.

Since I'm posting another message anyway, I guess I'll also point out that the "environment" section (i.e. client and server) could use some love. If I'm understanding correctly, Modrinth's database includes a, "client," and, "server," property for each mod, and each of those properties can independently be set to, "optional," "required," or, "unsupported." But right now, when you search the website, I think your only filtering options are, "any," and, "optional OR required." Plus, the client and server filters can only be combined with AND, not OR.

And in any case, it would be great to at least *tell* the user whether OR or AND is being used in a given context, so they wouldn't have to experiment or risk misunderstanding like I did here. Specifically: it's very confusing that categories and environments use AND, while loaders and Minecraft versions use OR, all within the same sidebar and without communicating this difference to the user at all.
